models/DataObject/ClassDefinition.php

```diff
@@ -348,7 +348,7 @@ public function save(bool $saveDefinitionFile = true): void
             $this->setId((string) $maxId);
         }
 
-        if (!preg_match('/[a-zA-Z][a-zA-Z0-9_]+/', $this->getName())) {
+        if (!preg_match('/[a-zA-Z]\w+/', $this->getName())) {
```

Comment:
It seems SonarCube wants something like
```suggestion
        if (!preg_match('/[a-zA-Z]\w+/u', $this->getName())) {
```